LuaMenu/widgets/gui_modoptions_panel.lua

```diff
@@ -37,10 +37,32 @@ local function UpdateControlValue(key, value)
 		elseif control.Select and control.itemKeyToName then -- combobox
 			control:Select(control.itemKeyToName[value])
 		elseif control.SetToggle then -- checkbox
-			control:SetToggle(value == true or value == 1)
+			control:SetToggle(value == true or value == 1 or value == "1")
 		end
 	end
 end
+
+local function TextFromNum(num, step)
+
+	-- remove excess accuracy
+	local places = 0
+	if step < 0.01  then
+		places = 3
+	elseif step < 0.1 then
+		places = 2
+	elseif step < 1 then
+		places = 1
+	end
+	local text = string.format("%." .. places .. "f", num)
+
+	-- remove trailing 0s
```

Comment:
I considered removing this trimming to fix step not being conveyed but decided it had likely been made so intentionally.